### PR TITLE
LPS-157838 - Always show FDS quick actions when actions menu is expanded

### PR DESCRIPTION
**Pull request by tinycarol**
### References

- [LPS-157838](https://issues.liferay.com/browse/LPS-157838)

### What is the goal?

* Keep FDS table row quick actions visible when actions drop-down is expanded even when row isn't hovered  

### What does it look like?

https://user-images.githubusercontent.com/6642927/202224409-3c39a657-8eeb-49ac-8f36-68e416851162.mov

### Notes
I don't like this solution because it adds an unnecessary re-render to the entire row when the drop-down is expanded, but since we need to use the state of that drop-down to show/hide another element of the row, with the current tools the only way is to move the state up to the common ancestor, in this case the row. There's alternatives (using a context maybe), but the one I like the most is this:

#### Alternative: use the `:has()` pseudo-class.

Currently the visibility of the quick actions is handled using CSS (SASS in this case):
```scss
.dnd-tr:hover:not(.nested):not(.selected) {
	.item-actions {
		overflow: visible;
		position: relative;
	}

	.quick-action-menu {
		background-color: inherit;
		display: flex;
		right: 48px;
	}

	.dnd-td:nth-last-child(2) {
		visibility: hidden;
	}
}
```
As you can see, this will apply the necessary styles when the row is being hovered.

Adding a second selector would do the same thing as the 65 lines of code in the PR:
```scss
.dnd-tr:hover:not(.nested):not(.selected),
.dnd-tr:has(button[aria-expanded="true"]):not(.nested):not(.selected) { 
	...
}
```

This will apply the nested styles to the table row when it contains an expanded drop-down. Just one line!

The downside is that, as we can see in [the caniuse page for `:has`](https://caniuse.com/css-has), it is a fairly recent spec that is only supported by **82.92%** of all users. **It's already available in all Chromium-based browsers**, but it's only available in Firefox under a feature flag (and with plenty of bugs) and in the tech preview of Safari.

Now, adding this code wouldn't break the page in the browsers that don't support `:has`, it would just stay the way it is now in those browsers (the quick actions would disappear if we hover a different row). So nothing actually breaks, but it results in inconsistencies between browsers.

I think we could add the simpler solution, since nothing breaks and it's only a matter of time that Firefox/Safari release this functionality. On the other hand, I don't know how this would affect QA tests, since I don't know if they run all tests in different browsers to check for compatibility.

TL;DR: We could replace all the changes in this PR with [this commit](https://github.com/liferay-frontend/liferay-portal/commit/3c9ca484bdf0de7b9fb7782eedf81b8e9adcef71), but with compat issues that don't break anything.

### CODE REVIEW

**Comment by mateusralv** [comment]
Hi @tinycarol the failures are not related to your PR 😄 

[comparison run](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1668631603769&testrayRunIdA=2054443870&testrayRunIdB=2057809985)

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js [hunk 1]
```diff
@@ -270,12 +271,48 @@ const ActionsCell = ({item, itemId, itemsActions}) => {
...
+	selected,
+	...props
+}) => {
+	const [menuActive, setMenuActive] = useState(false);
+
+	return (
+		<DndTable.Row
+			className={classNames(className, {
+				active,
+				expanded: menuActive,
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js:298]: Can we rename this class to `menu-active`? `expanded` seems a bit ambiguous.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js [hunk 2]
```diff
@@ -270,12 +271,48 @@ const ActionsCell = ({item, itemId, itemsActions}) => {
...
+	active,
+	children,
+	className,
+	item,
+	itemId,
+	itemsActions,
+	selected,
+	...props
+}) => {
+	const [menuActive, setMenuActive] = useState(false);
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js:292]: Nice setting of state scope! :sparkles:

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js [hunk 3]
```diff
@@ -270,12 +271,48 @@ const ActionsCell = ({item, itemId, itemsActions}) => {
...
+		>
+			{children}
+
+			<ActionsCell
+				item={item}
+				itemId={itemId}
+				itemsActions={itemsActions}
+				menuActive={menuActive}
+				onMenuActiveChange={setMenuActive}
+			/>
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js:311]: We can move `ActionCell` here inline. The only reason I moved `ActionCell` outside of main component is to reduce high amount of line breakage on deep indentation.

For this purpose, we could even move all of `{children}` here inline. It'll add some boilerpate with props, but it's probably worth it to make the code more readable.

**Review by markocikos — COMMENTED** [review]
@tinycarol Looks good! Some minor non-blocking stuff inline.

I did notice that styles of quick action menu seem a bit broken, it doesn't clearly overlap cells. But that's unrelated to this PR, we can fix it in a separate task.

**Comment by tinycarol** [comment]
ci:test:sf

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js (hunk 3 above)
Comment (tinycarol) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js]: Done! Moved `ActionCell` to inline and also all `{children}` elements. [Also changed `expanded` to `menu-active`](https://github.com/liferay-frontend/liferay-portal/pull/2866#discussion_r1030377652). See https://github.com/liferay-frontend/liferay-portal/pull/2866/commits/cc62e541f519255d3d41d5d20a403d21689765d1

**Comment by mateusralv** [comment]
ci:test:relevant

**Comment by mateusralv** [comment]
Hi @tinycarol the failures are not related to your PR.

**Review by markocikos — APPROVED** [review]
LGTM! :sparkles:

**Comment by markocikos** [comment]
ci:forward

**Comment by markocikos** [comment]
It should be easy for CI to pass PR that only affects FDS, and this is not an urgent fix. We can try standard forward once or twice before forwarding manually.
